app/controllers/deposits_controller.rb

```diff
@@ -63,6 +63,58 @@ def destroy
     end
   end
 
+  def get_endpoint(collection_slug,
+                   depositor_user_id)
+    case COLLECTIONS[:slug][collection_slug][:parser]
+    when "academic-commons"
+      Sword::Endpoints::AcademicCommonsEndpoint.new(collection_slug, depositor_user_id)
+    when "proquest"
+      Sword::Endpoints::ProquestEndpoint.new(collection_slug, depositor_user_id)
+    when "eprints"
+      Sword::Endpoints::EprintsEndpoint.new(collection_slug, depositor_user_id)
+    else
+      # raise an exception here
+    end
+  end
+
+  def resubmit
+    original_deposit = Deposit.find(params[:id])
+    endpoint = get_endpoint(original_deposit.collection_slug,
+                            original_deposit.depositor_user_id)
+
+    path_to_deposit_contents = original_deposit.content_path
+
+    # log basic essential info. Keep it terse! Gonna use :warn level, though not a warning.
+    Rails.logger.warn("About to resubmit deposit for deposit id: #{original_deposit.id}:" \
+                      "Collection slug: #{original_deposit.collection_slug}, " \
+                      "Username: #{original_deposit.depositor_user_id}, " \
+                      "Path to contents: #{path_to_deposit_contents}"
+                     )
+
+    endpoint.handle_deposit(path_to_deposit_contents)
+
+    # log basic essential info. Keep it terse! Gonna use :warn level, though not a warning.
+    Rails.logger.warn("Following is a re-deposit:" \
+                      "Title: #{endpoint.deposit_title.truncate_words(10)}, " \
+                      "Files: #{endpoint.documents_to_deposit}, " \
+                      "Hyacinth item pid: #{endpoint.adapter_item_identifier}, " \
+                      "Hyacinth asset pids: #{endpoint.asset_pids}, " \
+                      "Path to SWORD contents: #{path_to_deposit_contents}"
+                     )
+
+    # create Deposit instance to store deposit info in database
+    resubmit_deposit = Deposit.new
```

Comment:
in the interest of keeping the controller slim, this deposit creation logic could be moved to the deposit model

Reply:
I did consider that option. I've opened a ticket, SWORD-119, and I will investigate (and possibly implement) that option.